control-plane/api/v1alpha1/serviceintentions_types.go

```diff
@@ -244,8 +244,8 @@ func (in *ServiceIntentions) MatchesConsul(candidate api.ConfigEntry) bool {
 	return cmp.Equal(
 		in.ToConsul(""),
 		configEntry,
-		cmpopts.IgnoreFields(capi.ServiceIntentionsConfigEntry{}, "Partition", "Namespace", "Meta", "ModifyIndex", "CreateIndex"),
-		cmpopts.IgnoreFields(capi.SourceIntention{}, "Partition", "Namespace", "LegacyID", "LegacyMeta", "LegacyCreateTime", "LegacyUpdateTime", "Precedence", "Type"),
+		cmpopts.IgnoreFields(capi.ServiceIntentionsConfigEntry{}, "Partition", "Meta", "ModifyIndex", "CreateIndex"),
+		cmpopts.IgnoreFields(capi.SourceIntention{}, "LegacyID", "LegacyMeta", "LegacyCreateTime", "LegacyUpdateTime", "Precedence", "Type"),
```

Comment:
Did you happen to test this against OSS consul? I remember that we ignored these fields because consul OSS will default them and in kube they'd be empty.

Reply:
Ahh, that makes sense. Nice catch.

I just updated this check to consider `"default"` and `""` the same. Just to make sure, I also tested this out and am not seeing an infinite reconcile loop.

Here are the relevant logs from testing this out a few times:
```
2023-05-29T20:40:18.003Z [ERROR] agent.http: Request error: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 error="Config entry not found for \"service-intentions\" / \"static-server\""
2023-05-29T20:40:18.003Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="359.767µs"
2023-05-29T20:40:18.006Z [DEBUG] agent.http: Request finished: method=PUT url=/v1/config?dc=dc1 from=10.42.0.14:36534 latency=2.449821ms
2023-05-29T20:40:18.094Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="306.522µs"
2023-05-29T20:40:36.485Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="130.518µs"
2023-05-29T20:40:36.488Z [DEBUG] agent.http: Request finished: method=PUT url=/v1/config?dc=dc1 from=10.42.0.14:36534 latency=1.921819ms
2023-05-29T20:40:36.500Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="162.379µs"
2023-05-29T20:41:00.852Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="146.442µs"
2023-05-29T20:42:28.201Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="147.631µs"
2023-05-29T20:42:28.205Z [DEBUG] agent.http: Request finished: method=PUT url=/v1/config?dc=dc1 from=10.42.0.14:36534 latency=2.592968ms
2023-05-29T20:42:28.213Z [DEBUG] agent.http: Request finished: method=GET url=/v1/config/service-intentions/static-server?dc=dc1 from=10.42.0.14:36534 latency="166.749µs"
```